### PR TITLE
Reworked proof of work 

### LINKED ISSUE #288: Rework proof of work interface to specify number of bits
Current implementation accepts mask.

### PR DESCRIPTION
**Pull request by vo-nil**
closes #288

### CODE REVIEW

File: libs/algebra/test/multiexp.cpp [hunk 1]
```diff
@@ -158,11 +158,12 @@ BOOST_AUTO_TEST_SUITE(multiexp_test_suite)
 
 BOOST_AUTO_TEST_CASE(multiexp_test_case) {
 
+    // TODO: These tests look like performance benchmarks. Should be removed from CI
     std::cout << "Testing BLS12-381 G1" << std::endl;
-    print_performance_csv<curves::bls12<381>::g1_type<>, curves::bls12<381>::scalar_field_type>(2, 12, 14, true);
+    print_performance_csv<curves::bls12<381>::g1_type<>, curves::bls12<381>::scalar_field_type>(2, 4, 6, true);
```
Comment (martun) [inline libs/algebra/test/multiexp.cpp]: Maybe we move this to bench_test folder?

File: libs/zk/include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp [hunk 2]
```diff
@@ -44,37 +45,40 @@ namespace nil {
                     using transcript_type = transcript::fiat_shamir_heuristic_sequential<transcript_hash_type>;
                     using output_type = OutType;
 
-                    static inline OutType generate(transcript_type &transcript, OutType mask=0xFFFF) {
+                    static inline std::array<std::uint8_t, sizeof(OutType)>
+                        int_be(OutType v) {
```
Comment (martun) [inline libs/zk/include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp]: Maybe rename this to something like to_byte_array() ?

**Review by martun — APPROVED** [review]

File: libs/zk/include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp (hunk 2 above)
Comment (vo-nil) [reply libs/zk/include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp]: renamed

File: libs/algebra/test/multiexp.cpp (hunk 1 above)
Comment (vo-nil) [reply libs/algebra/test/multiexp.cpp]: moved to benches

**Pull request merged**